node/node.go

```diff
@@ -410,24 +411,27 @@ func (nc *Config) Build(ctx context.Context) (*Node, error) {
 		nodeConsensus = consensus.NewExpected(&cstOffline, bs, processor, powerTable, genCid, nc.Verifier)
 	}
 
-	chainFacade := bcf.NewBlockChainFacade(chainStore, &cstOffline)
-
-	// only the syncer gets the storage which is online connected
-	chainSyncer := chain.NewDefaultSyncer(&cstOffline, nodeConsensus, chainStore, fetcher)
-	msgPool := core.NewMessagePool(chainStore, nc.Repo.Config().Mpool, consensus.NewIngestionValidator(chainFacade, nc.Repo.Config().Mpool))
-	outbox := core.NewMessageQueue()
-
-	// Set up libp2p pubsub
+	// Set up libp2p network
 	fsub, err := libp2pps.NewFloodSub(ctx, peerHost)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to set up pubsub")
+		return nil, errors.Wrap(err, "failed to set up network")
 	}
+
 	backend, err := wallet.NewDSBackend(nc.Repo.WalletDatastore())
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to set up wallet backend")
 	}
 	fcWallet := wallet.New(backend)
 
+	// only the syncer gets the storage which is online connected
+	chainSyncer := chain.NewDefaultSyncer(&cstOffline, nodeConsensus, chainStore, fetcher)
+	msgPool := core.NewMessagePool(chainStore, nc.Repo.Config().Mpool, consensus.NewIngestionValidator(chainFacade, nc.Repo.Config().Mpool))
+	msgQueue := core.NewMessageQueue()
+
+	msgPublisher := newDefaultPublisher(pubsub.NewPublisher(fsub), core.Topic, msgPool)
+	actorProvider := newDefaultActorProvider(chainStore, &cstOffline)
```

Comment:
The publisher and actor provider are the "wiring" between the outbox and other components.

Reply:
The `ActorProvider` looks a lot like this existing `chainFacade` plumbing. Maybe we should combine the `chainProvider` and `actorProvider` interfaces on `Outbox` and replace them both with `chainFacade`?

On a somewhat related note, I'm thinking we should also extract `chainFacade` from plumbing later on too.

Reply:
Also, btw `chainFacade` was a terrible choice of name on my part, and I would totally be up for calling it something like `actorChainDecorator` or `chainStateProvider` instead.

Reply:
Also, btw `chainFacade` was a terrible choice of name on my part, and I would totally be up for calling it something like `actorChainDecorator` or `chainStateProvider` instead.

Reply:
I've replaced the actorProvider with a new method on the chain facade (which differs from the existing one in being parameterised by tipset key). I also renamed it to `ChainStateProvider`.

I agree that it should move moved from plumbing. Some of the methods are just compositions of lower-level bits, so it should probably go in porcelain (with a few primitives remaining behind).

I'd rather not combine the chain and actor interfaces from the outbox, since they are provided by different objects at the moment.